core/src/main/java/org/apache/iceberg/ClientPool.java

```diff
@@ -26,4 +28,16 @@
   <R> R run(Action<R, C, E> action) throws E, InterruptedException;
 
   <R> R run(Action<R, C, E> action, boolean retry) throws E, InterruptedException;
+
+  /**
+   * Initialize the client pool with catalog properties and an optional hadoop configuration.
```

Comment:
nit: do we need 'optional' to describe hadoop configuration?  (its always passed in, so seems redundant to say this)